### PR TITLE
Fix InvalidTransactionNativeTokensCount error for multiple entries

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Fix InvalidTransactionNativeTokensCount error for multiple entries with the same native token

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Added tests and getting this block https://api.shimmer.network/api/core/v2/blocks/0x8ff7d61fae6687303aac3c2a47b1d6599b35428d2ca2296ebc25e5856ae90f4b with an example

### CODE REVIEW

File: sdk/CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 
 - `needs_blind_signing()` for non Ed25519 addresses;
+- InvalidTransactionNativeTokensCount error for multiple entries with the same native token;
```
Comment (thibault-martinez) [inline sdk/CHANGELOG.md]:
```suggestion
- `InvalidTransactionNativeTokensCount` error for multiple entries with the same native token;
```

File: sdk/src/types/block/payload/transaction/essence/regular.rs [hunk 2]
```diff
@@ -219,7 +219,7 @@ fn verify_inputs_packable<const VERIFY: bool>(inputs: &[Input], _visitor: &Proto
 fn verify_outputs<const VERIFY: bool>(outputs: &[Output], visitor: &ProtocolParameters) -> Result<(), Error> {
     if VERIFY {
         let mut amount_sum: u64 = 0;
-        let mut native_tokens_count: u8 = 0;
+        let mut total_native_tokens: HashSet<TokenId> = HashSet::new();
```
Comment (thibault-martinez) [inline sdk/src/types/block/payload/transaction/essence/regular.rs]:
```suggestion
        let mut total_native_tokens = HashSet::<TokenId>::new();
```
Less repetitive

**Review by thibault-martinez — DISMISSED** [review]

File: sdk/tests/types/transaction_regular_essence.rs [hunk 3]
```diff
@@ -470,3 +470,63 @@ fn duplicate_output_foundry() {
...
     ));
 }
+
+#[test]
+fn more_than_64_same_native_tokens() {
+    let protocol_parameters = protocol_parameters();
+    let transaction_id = TransactionId::new(prefix_hex::decode(TRANSACTION_ID).unwrap());
+    let input = Input::Utxo(UtxoInput::new(transaction_id, 0).unwrap());
+    let bytes: [u8; 32] = prefix_hex::decode(ED25519_ADDRESS_1).unwrap();
+    let address = Address::from(Ed25519Address::new(bytes));
```
Comment (Alex6323) [inline sdk/tests/types/transaction_regular_essence.rs]: there's a `FromStr` impl for `Ed25519Address`, right?

File: sdk/tests/types/transaction_regular_essence.rs [hunk 4]
```diff
@@ -470,3 +470,63 @@ fn duplicate_output_foundry() {
...
+
+    let essence = RegularTransactionEssence::builder(protocol_parameters.network_id(), rand_inputs_commitment())
+        .with_inputs([input])
+        .with_outputs(outputs)
+        .finish_with_params(&protocol_parameters);
+
+    assert!(matches!(
+        essence,
+        Err(Error::InvalidTransactionNativeTokensCount(count)) if count == 65
+    ));
```
Comment (Alex6323) [inline sdk/tests/types/transaction_regular_essence.rs:517]: can we remove the extra binding?

File: sdk/tests/types/transaction_regular_essence.rs [hunk 5]
```diff
@@ -470,3 +470,63 @@ fn duplicate_output_foundry() {
...
+        .add_unlock_condition(AddressUnlockCondition::new(address))
+        .finish_output(protocol_parameters.token_supply())
+        .unwrap();
+
+    let essence = RegularTransactionEssence::builder(protocol_parameters.network_id(), rand_inputs_commitment())
+        .with_inputs([input])
+        .with_outputs(vec![basic; 100])
+        .finish_with_params(&protocol_parameters);
+
+    assert!(essence.is_ok());
```
Comment (Alex6323) [inline sdk/tests/types/transaction_regular_essence.rs:484]: can we remove the extra binding?

File: sdk/tests/types/transaction_regular_essence.rs (hunk 4 above)
Comment (Thoralf-M) [reply sdk/tests/types/transaction_regular_essence.rs:517]: I think that would be less clear and we also have it like this for all the tests above

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**